gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/DiagnosticReportMapper.java

```diff
@@ -0,0 +1,203 @@
+            .collect(toCollection(ArrayList::new));
+
+        if (!resultReferences.isEmpty()) {
+            diagnosticReport.setResult(resultReferences);
+        }
+    }
+
+    private Optional<Observation> getObservationCommentById(List<Observation> observationComments, String id) {
+        return observationComments.stream()
+            .filter(observationComment -> id.equals(observationComment.getId()))
+            .findFirst();
+    }
+
+    private Optional<Identifier> createIdentifierExtension(List<II> id) {
+        if (id.size() > 1) {
+            final II idExtension = id.get(1);
+            if (idExtension != null && EXTENSION_IDENTIFIER_ROOT.equals(idExtension.getRoot())) {
+                return Optional.of(new Identifier()
+                    .setSystem(EXTENSION_IDENTIFIER_ROOT)
+                    .setValue(idExtension.getExtension()));
+            }
+        }
+        return Optional.empty();
+    }
+
+    private Optional<Reference> buildContext(List<RCMRMT030101UK04EhrComposition> compositions, List<Encounter> encounters,
+        RCMRMT030101UK04CompoundStatement compoundStatement) {
+        var currentEhrComposition = getCurrentEhrComposition(compositions, compoundStatement);
+        return encounters.stream()
+            .filter(encounter -> encounter.getId().equals(currentEhrComposition.getId().getRoot()))
+            .findFirst()
+            .map(Reference::new);
+    }
+
+    private Optional<InstantType> getIssued(RCMRMT030101UK04EhrExtract ehrExtract, List<RCMRMT030101UK04EhrComposition> compositions,
+        RCMRMT030101UK04CompoundStatement compoundStatement) {
+        var ehrComposition = getCurrentEhrComposition(compositions, compoundStatement);
+
+        if (authorHasValidTimeValue(ehrComposition.getAuthor())) {
+            return Optional.of(parseToInstantType(ehrComposition.getAuthor().getTime().getValue()));
+        }
+
+        if (availabilityTimeHasValue(ehrExtract.getAvailabilityTime())) {
+            return Optional.of(parseToInstantType(ehrExtract.getAvailabilityTime().getValue()));
+        }
+
+        return Optional.empty();
+    }
+
+    private boolean authorHasValidTimeValue(RCMRMT030101UK04Author author) {
+        return author != null && author.hasTime()
+            && author.getTime().hasValue()
+            && !author.getTime().hasNullFlavor();
+    }
+
+    private boolean availabilityTimeHasValue(TS availabilityTime) {
+        return availabilityTime != null && availabilityTime.hasValue() && !availabilityTime.hasNullFlavor();
+    }
+
+    private CodeableConcept createCodeableConcept(RCMRMT030101UK04CompoundStatement compoundStatement) {
+        return codeableConceptMapper.mapToCodeableConcept(compoundStatement.getCode());
+    }
+
+    private List<RCMRMT030101UK04EhrComposition> getCompositionsContainingClusterCompoundStatement(RCMRMT030101UK04EhrExtract ehrExtract) {
```

Comment:
Should be able to remove these last two methods with the abstract mapper implementation